itests/eth_fee_history_test.go

```diff
@@ -29,95 +46,109 @@ func TestEthFeeHistory(t *testing.T) {
 	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
 	defer cancel()
 
-	// Wait for the network to create 20 blocks
+	heads, err := client.ChainNotify(ctx)
+	require.NoError(err)
+
+	// Save the full view of the tipsets to calculate the answer when there are null rounds
+	tsHeights := []int{1}
+	go func() {
+		for chg := range heads {
+			for _, c := range chg {
+				tsHeights = append(tsHeights, int(c.Val.Height()))
+			}
+		}
+	}()
+
+	miner[0].InjectNulls(abi.ChainEpoch(1))
+	<-time.After(5 * blockTime)
+	miner[0].InjectNulls(abi.ChainEpoch(3))
+	<-time.After(2 * blockTime)
+	miner[0].InjectNulls(abi.ChainEpoch(1))
+
+	// Wait for the network to create at least 20 blocks
 	client.WaitTillChain(ctx, kit.HeightAtLeast(20))
 	for _, m := range miner {
 		m.Pause()
 	}
+	<-time.After(5 * blockTime)
```

Comment:
This is racy. It's possible that the miners may have not pausd yet. Safer to do a ChainNotify with a 5 second timeout and assert that it errors with context deadline exceeded.

Reply:
Hm, that won't work, but we should wait for 5 seconds of ChainNotify not sending any notifications.

Reply:
Honestly I think this is fine, just keep it as it is. Pause is synchronous, so we're just waiting on draining all pending chain notifications, and 5 seconds is of plenty time for that.